judgments/views/detail.py

```diff
@@ -35,9 +35,11 @@ class NoNeutralCitationError(Exception):
     logging.getLogger("weasyprint").handlers = []
 
 
-def get_published_document_by_uri(document_uri: str) -> Document:
+def get_published_document_by_uri(document_uri: str, cache_if_not_found: bool = False) -> Document:
     try:
-        document = get_document_by_uri(document_uri)
+        document = get_document_by_uri(document_uri, cache_if_not_found=cache_if_not_found)
```

Comment:
Pondering whether there are security issues with the cache, since different users have different access -- no, because the editor UI and public UI are completely isolated.

Reply:
I did wonder this - there's potentially a max 15 minute window where a document which was previously published and has been yanked would still be visible, but I'm not unduly worried about this.

In due course we should add some kind of API to the PUI which lets us trigger things like cache flushing based on EUI interactions.